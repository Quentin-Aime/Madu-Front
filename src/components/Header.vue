<template>
  <div class="header">
    <div class="logo">
      <img src="" alt="MADU logo" />
    </div>
    <div class="interactions">
      <el-input class="search-input" v-model="searchFilter"></el-input>
      <v-icon name="power-off"></v-icon>
    </div>
  </div>
</template>

<script>
import "vue-awesome/icons/power-off";

export default {
  components: {},

  props: {},

  data: function() {
    return {
      searchFilter: ""
    };
  },

  computed: {},

  mounted: function() {},

  methods: {}
};
</script>

<style lang="scss">
.header {
  display: flex;
  align-items: center;
  .logo {
    width: 10%;
  }
  .interactions {
    width: 90%;
    display: flex;
    justify-content: space-around;
    align-items: center;
    .search-input {
      input {
        width: 100px;
        border-radius: 30px;
        transition: all ease-in-out 0.25s;
        &:focus {
          width: 200px;
        }
      }
    }
  }
}
</style>
