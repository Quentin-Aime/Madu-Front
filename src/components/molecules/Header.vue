<template>
  <div class="header">
    <div class="logo">
      <img src="" alt="MADU logo" />
    </div>
    <div class="interactions">
      <el-input class="search-input" v-model="searchFilter"></el-input>
      <div class="logout-icon">
        <v-icon name="power-off"></v-icon>
      </div>
    </div>
  </div>
</template>

<script>
import "vue-awesome/icons/power-off";

export default {
  components: {},

  props: {},

  data: function() {
    return {
      searchFilter: ""
    };
  },

  computed: {},

  mounted: function() {},

  methods: {}
};
</script>

<style lang="scss">
.header {
  display: flex;
  align-items: center;
  padding: 20px 0;
  border-bottom: 1px solid #CCC;
  .logo {
    width: 20%;
  }
  .interactions {
    width: 80%;
    display: flex;
    justify-content: space-around;
    align-items: center;
    padding: 0 30px;
    .search-input {
      input {
        width: 200px;
        border-radius: 30px;
        transition: all ease-in-out 0.25s;
        border: 1px solid #ccc;
        &:focus {
          width: 400px;
        }
      }
    }
    .logout-icon {
      border-radius: 30px;
      padding: 11px 11px 5px 11px;
      border: 1px solid #ccc;
    }
  }
}
</style>
