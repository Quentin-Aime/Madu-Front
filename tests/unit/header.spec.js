import { shallowMount } from "@vue/test-utils";
import Header from "../../src/components/molecules/Header.vue";

it("should match snapshot", () => {
  const wrapper = shallowMount(Header);

  expect(wrapper.element).toBeDefined();
  expect(wrapper.element).toMatchSnapshot();
});
