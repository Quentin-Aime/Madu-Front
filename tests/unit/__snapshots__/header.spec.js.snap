// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  class="header"
>
  <h1
    class="title"
  >
    MADU
  </h1>
   
  <el-button
    class="logout-button"
    type="primary"
  >
    Se d√©connecter
  </el-button>
</div>
`;
